stl/inc/memory

```diff
@@ -4246,20 +4246,22 @@ public:
             _Smart_ptr.release();
         }
 
-        if (!_Get_ptr()) {
-            return;
-        }
-
         _STD apply(
             [this](auto&&... _Args_) {
                 using _Sp = _Pointer_of_or<_SmartPtr, _Pointer>;
                 if constexpr (is_pointer_v<_SmartPtr>) {
                     _Smart_ptr = _SmartPtr(static_cast<_Sp>(_Get_ptr()), _STD forward<_ArgsT>(_Args_)...);
-                } else if constexpr (_Resettable_pointer<_SmartPtr, _Sp, _Pointer, _ArgsT...>) {
-                    _Smart_ptr.reset(static_cast<_Sp>(_Get_ptr()), _STD forward<_ArgsT>(_Args_)...);
                 } else {
-                    static_assert(is_constructible_v<_SmartPtr, _Sp, _ArgsT...>, "(N4950 [inout.ptr.t]/11.4)");
-                    _Smart_ptr = _SmartPtr(static_cast<_Sp>(_Get_ptr()), _STD forward<_ArgsT>(_Args_)...);
+                    if (!_Get_ptr()) {
+                        return;
+                    }
+
+                    if constexpr (_Resettable_pointer<_SmartPtr, _Sp, _Pointer, _ArgsT...>) {

```

Comment:
I believe that it would be simpler to move this `if (!_Get_ptr())` into the initial `if constexpr (!is_pointer_v<_SmartPtr>)` branch. This will more clearly coalesce the *`release-statement;`* and `if (p) {` in the Standardese, as well as imitating how it skips calling `apply()` for null non-raw pointers.

(As a bonus it also dramatically simplifies the diff, but that's not the main reason why it's desirable.)